rsocket-core/src/main/java/io/rsocket/RSocketRequester.java

```diff
@@ -200,233 +195,248 @@ public boolean isDisposed() {
   }
 
   private Mono<Void> handleFireAndForget(Payload payload) {
-    return lifecycle.active(
-        () -> {
-          final int streamId = streamIdSupplier.nextStreamId();
-          ByteBuf requestFrame =
-              RequestFireAndForgetFrameFlyweight.encode(
-                  allocator,
-                  streamId,
-                  false,
-                  payload.hasMetadata() ? payload.sliceMetadata().retain() : null,
-                  payload.sliceData().retain());
-          payload.release();
-          sendProcessor.onNext(requestFrame);
-        });
+    Throwable err = checkAvailable();
+    if (err != null) {
+      return Mono.error(err);
+    }
+
+    final int streamId = streamIdSupplier.nextStreamId();
+    ByteBuf requestFrame =
+        RequestFireAndForgetFrameFlyweight.encode(
+            allocator,
+            streamId,
+            false,
+            payload.hasMetadata() ? payload.sliceMetadata().retain() : null,
+            payload.sliceData().retain());
+    payload.release();
+    sendProcessor.onNext(requestFrame);
+
+    return Mono.empty();
   }
```

Comment:
The request* methods should probably surrounded by a try/catch that catches a throwable so if there is an exception it can be handled.

Like:

`Throwable err = checkAvailable();
    if (err != null) {
      return Mono.error(err);
    }

    try { 
    int streamId = streamIdSupplier.nextStreamId();
    final UnboundedProcessor<ByteBuf> sendProcessor = this.sendProcessor;
    final ByteBuf requestFrame =
        RequestResponseFrameFlyweight.encode(
            allocator,
            streamId,
            false,
            payload.sliceMetadata().retain(),
            payload.sliceData().retain());

    payload.release();

    UnicastMonoProcessor<Payload> receiver = UnicastMonoProcessor.create();
    receivers.put(streamId, receiver);

    sendProcessor.onNext(requestFrame);
    return receiver
        .doOnError(t -> sendProcessor.onNext(ErrorFrameFlyweight.encode(allocator, streamId, t)))
        .doFinally(
            s -> {
              if (s == SignalType.CANCEL) {
                sendProcessor.onNext(CancelFrameFlyweight.encode(allocator, streamId));
              }

              receivers.remove(streamId);
       });
  } catch (Throwable t) {
    throw Exceptions.propogate(t);
  }

`

Reply:
Agree. There is another question after looking at this yesterday and now: if `defer` is removed completely, and `singleSubscriberRSocket` enabled in RSocketFactory, not only termination and lease check are performed eagerly, but frames are sent on call and not subscription.

Reply:
So I think I'll send frames in `receiver`'s onSubscribe?

Reply:
Sending `request` frame on call may be preferable way for single subscriber `RequestRSocket`, but this behavior should be consistent across all interactions (e.g. requestStream sends on subscription)